dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -149,6 +149,35 @@ def delete_progress_form
       count
   end
 
+  def delete_progress
+    params.require([:user_id, :script_id, :reason])
+
+    # TODO: Check for escaping issues for 'reason' once everything is hooked up end-to-end.
+
+    user_id = params[:user_id]
+    script_id = params[:script_id]
+    user_storage_id = storage_id_for_user_id(user_id)
+
+    FirehoseClient.instance.put_record(
+      :analysis,
+      {
+        study: 'reset-progress',
+        event: 'admin-delete-progress',
+        user_id: user_id,
+        script_id: script_id,
+        data_json: {
+          signed_in_user: current_user.username,
+          reason: params[:reason]
+        }.to_json
+      }
+    )
+
+    UserScript.where(user_id: user_id, script_id: script_id).destroy_all
```

Comment:
For my learning, what is the user script table for?

Reply:
The user_scripts table tracks which scripts have user progress.  It is the table that's [used to decide](https://github.com/code-dot-org/code-dot-org/blob/aceebfe4f2a633bef861cd5a6afa602d79d58973/dashboard/app/models/user.rb#L1645) which cards to show in the student homepage via some activerecord [magic](https://github.com/code-dot-org/code-dot-org/blob/aceebfe4f2a633bef861cd5a6afa602d79d58973/dashboard/app/models/user.rb#L438).  We [write to this table](https://github.com/code-dot-org/code-dot-org/blob/aceebfe4f2a633bef861cd5a6afa602d79d58973/dashboard/app/models/user.rb#L1838-L1840) in the main track progress codepath.